sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/parquet/ParquetRowConverter.scala

```diff
@@ -176,10 +177,22 @@ private[parquet] class ParquetRowConverter(
    */
   def currentRecord: InternalRow = currentRow
 
+
   // Converters for each field.
   private[this] val fieldConverters: Array[Converter with HasParentContainerUpdater] = {
+    // For letter case issue, create name to field index based on case sensitivity
+    val catalystFieldNameToIndex = if (caseSensitive) {
+      catalystType.fieldNames.zipWithIndex.toMap
+    } else {
+      CaseInsensitiveMap(catalystType.fieldNames.zipWithIndex.toMap)
+    }
+
     parquetType.getFields.asScala.map { parquetField =>
```

Comment:
@kimtkyeom, can we add an if-else on `spark.sql.optimizer.nestedSchemaPruning.enabled`, and keep the very original codes https://github.com/apache/spark/pull/24307/files#diff-d07d3850a0ad8635a0c5562ed1e8bf26L185-L188?